docs/understanding-airbyte/schemaless-sources-and-destinations.md

```diff
@@ -0,0 +1,53 @@
+Similarly, for a file-based source such as S3, the files that are present in your source may not all have the same schema.
+
+Although the sources themselves may not conform to an obvious schema, Airbyte still needs to know the shape of the data in order to prepare the destination for the records.
+For these sources, during the [`discover`](/understanding-airbyte/airbyte-protocol#discover) method, Airbyte offers two options to create the schema:
+
+1. Dynamic schema inference.
+2. A hardcoded "schemaless" schema.
+
+### Dynamic schema inference
+If this option is selected, Airbyte will infer the schema dynamically based on the contents of the source.  If your source's content is homogenous, we recommend this option, as the data in your destination will be typed and you can make use of schema evolution features, column selection, and similar Airbyte features which operate against the source's schema. 
+For MongoDB, if the dataset exceeds 10,000 documents, the schema is inferred from a random sample of up to 10,000 documents; otherwise, it is inferred from the entire dataset.
+For file-based sources, we look at up to 10 files (reading up to 1MB per file) and infer the schema based on the contents of those files.
+In both cases, as the contents of the source change, the schema can change too.
+
+The schema that's produced from the inference procedure will include all the top-level fields that were observed in the sampled records.
+The type assigned to each field will be the widest type observed for that field in any of the sampled data.
+So if we observe that a field has an integer type in one record and a string in another, the schema will identify the field as a string.
+
+There are a few drawbacks to be aware of:
+- If your dataset is very large, the `discover` process can be very time-consuming.
+- Because we may not use 100% of the available data to create the schema, your schema may not contain every field present in your records.
+  Airbyte only syncs fields that are in the schema, so you may end up with incomplete data in the destination.
+
+If your data set is very large or you anticipate that it will change often, we recommend using the "schemaless" schema.
+
+### Schemaless schema
+If this option is selected, the schema will always be `{"data": object}`, regardless of the contents of the data. During the sync, we "wrap" each record behind a key named `data`. This means that the destination receives the data with one top-level field only, and the value of the field is the entire record.
+This option avoids a time-consuming or inaccurate `discover` phase and guarantees that everything ends up in your destination, at the expense of Airbyte being able to structure the data into different columns.
+
+## Future Enhancements
```

Comment:
Suggestion: a way to exclude fields from syncing in schemaless also, as column selection is not applicable.
This is on the radar for mongodb.

Reply:
👍 added.